extensions/amp-analytics/0.1/requests.js

```diff
@@ -133,10 +144,46 @@ export class RequestHandler {
 
     if (!this.baseUrlPromise_) {
       expansionOption.freezeVar('extraUrlParams');
-      this.baseUrlTemplatePromise_ = this.variableService_.expandTemplate(
-        this.baseUrl,
-        expansionOption
-      );
+
+      // expand requestOrigin if it is declared
+      if (this.requestOrigin) {
+        const baseUrlTemplatePromise = this.variableService_.expandTemplate(
+          this.baseUrl,
+          expansionOption
+        );
+
+        // do not encode vars in request origin
+        expansionOption.noEncode = true;
```

Comment:
I'm afraid that `expandTemplate` function is an async function, that the `expansionOption.noEncode` is changed  before it's used.

Reply:
Solved by creating a new copy of ExpansionOptions to use in request origin expansion